### DISCUSSION TITLE
ObservationValue Type in ObxSegment

### DISCUSSION THREAD

**Discussion by scavet64** [davebronson/clear-hl7-net#8] [General]
I was curious why the `ObservationValue` in the class `ObxSegment` for v2.5.1 and v2.6.0 are of type `ChannelDefinition` while the rest of the classes have that property as type `IEnumerable<string>`? 

To my understanding, that field's type can vary depending on the value in OBX.2 according to the spec. With that in mind I was thinking that changing `ObservationValue`'s data type to IType would resolve that issue. Would that change be inline with your thinking as well?

I have only recently started working with the HL7 format and thanks to your library I've been able to piece it together and get a better grasp so thanks again!

**Comment by davebronson** [comment]
Hey @scavet64, thank you for the comments.  Glad that my library is helping!

In regards to the `ObxSegment.ObservationValue` property, thank you for bringing this to my attention.  I dug back into the spec documents from HL7, and here's what I found:

Version | Spec - OBX.5 | clear-hl7-net Uses | Comments
------------ | ------------- | ------------- | -------------
2.3 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.31 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.4 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.5 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.51 | Variable, Repeatable | `ChannelDefinition` | Incorrect.<br />clear-hl7-net needs to be<br />changed to `IEnumerable<string>`.<br />**Breaking Change**
2.6 | Channel Definition (CD), Not Repeatable | `ChannelDefinition` | While clear-hl7-net matches<br />HL7's spec, I'm questioning the accuracy<br />of the spec document.<br />**Further verification needed**.
2.7 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.8 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.81 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.82 | Variable, Repeatable | `IEnumerable<string>` | ✓
2.9 | Variable, Repeatable | `IEnumerable<string>` | ✓

I'll add an Issue to fix for V2.51.

Do you have any references that would help clear up for V2.6?  I sometimes refer to Caristix, though I can't really comment on how reliable their HL7 definition is.  [https://hl7-definition.caristix.com/v2/HL7v2.6/Segments/OBX](url)

**Comment by davebronson** [comment]
I've created a new issue for V2.51 item:  #9

**Comment by davebronson** [comment]
I referred to the Medical Records/Information Management specification for 2.6, and it also calls for a variable datatype.  Not ChannelDefinition, as the Appendix document claims.  This matches with every other OBX reference from V2.6 that I checked against on the web.  So I'm going to update clear-hl7-net to use `IEnumerable<string>` for V2.6 as well.

**Comment by davebronson** [comment]
I've created a new issue for V2.60 item: #10

**Comment by davebronson** [comment]
Changes now implemented in the v2-enhancements branch.

**Comment by davebronson** [comment]
v2-beta1 has now been published to NuGet with the `ObxSegment.ObservationValue` changes in place.  @scavet64
